plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/editor/context/project/ProjectContextProvider.kt

```diff
@@ -264,59 +262,35 @@ class ProjectContextProvider(val project: Project, private val encoderServer: En
     }
 
     private fun isBuildOrBin(filePath: String): Boolean {
-        val regex = Regex("""[/\\](bin|build|node_modules|env|\.idea)[/\\]""", RegexOption.IGNORE_CASE)
+        val regex = Regex("""[/\\](bin|build|node_modules|venv|env|\.idea)[/\\]""", RegexOption.IGNORE_CASE)
         return regex.find(filePath) != null
     }
 
-    // TODO: this code needs to be refactored and shared with code scan
-    @Suppress("LoopWithTooManyJumpStatements")
     private fun collectFiles(): FileCollectionResult {
-        val files = mutableSetOf<String>()
-        val traversedDirectories = mutableSetOf<VirtualFile>()
+        val collectedFiles = mutableListOf<String>()
         var currentTotalFileSize = 0L
-        val stack = Stack<VirtualFile>()
-        moduleLoop@ for (module in project.modules) {
-            if (module.guessModuleDir() != null) {
-                stack.push(module.guessModuleDir())
-                while (stack.isNotEmpty()) {
-                    val current = stack.pop()
+        val changeListManager = ChangeListManager.getInstance(project)
+        val featureDevSessionContext = FeatureDevSessionContext(project)
+        val allFiles = project.guessProjectDir()?.let {
+            VfsUtil.collectChildrenRecursively(it).filter { child ->
+                val language = child.programmingLanguage()
+                !changeListManager.isIgnoredFile(child) && !isBuildOrBin(child.path) &&
+                    runBlocking { !featureDevSessionContext.ignoreFile(child, scope) } && child.length <= 10 * 1024 * 1024 &&
+                    language != CodeWhispererPlainText.INSTANCE && language != CodeWhispererUnknownLanguage.INSTANCE
+            }
```

Comment:
nit
```suggestion
                !changeListManager.isIgnoredFile(child) && 
                !isBuildOrBin(child.path) &&
                 runBlocking { !featureDevSessionContext.ignoreFile(child, scope) } && 
                 child.length <= 10 * 1024 * 1024 &&
                 language !is CodeWhispererPlainText && 
                 language !is CodeWhispererUnknownLanguage
            }
```

Reply:
the index actually supports multiple languages include Config ( brazil config), etc that may not be CodeWhisperer inline completion languages

Reply:
ok then we don't need to check for language, I can take it out